### DISCUSSION TITLE
Simple (modal) resources: route to edit

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#4861] [Help]
Hello,

I'm unable to find a way to programmatically open a modal to edit a record on simple modal resources, i.e., having a link that renders the resource table and opens the edit modal for a given record. Is this possible?

I've tried adding a param with the record to edit to the url, and then playing with the mount() function on the `ManageXXXX` class, mounting the action the same way clicking on an edit button does, but that doesn't work because at this point `Filament\Resources\Pages\ListRecords::$cachedTableActions` is not initialized.

I can make a PR to make this available, but would need some guidance on how to achieve this. This could also be helpful to have a route to directly open the modal for creating a new record.

**Comment by danharrin** [comment]
Unfortunately it's not possible, not because of that uninitialised property but because Livewire doesnt allow you to dispatch browser events on the initial request. Maybe in v3?

**Reply by marc31** [reply]
Hello,

Do you think is now possible ?

**Reply by danharrin** [reply]
Livewire still doesn't allow it, but we may implement a workaround at some point

**Reply by frknasir** [reply]
What about now? Is it possible?

**Reply by howdu** [reply]
Yes possible with v4+ pass the record ID as params
```
ProductResource::getUrl(parameters: [
    'tableAction' => 'edit',
    'tableActionRecord' => 123,
])
```
